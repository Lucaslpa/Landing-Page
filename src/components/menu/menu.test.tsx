import { render } from 'utils/test-utils';
import { screen, fireEvent } from '@testing-library/react';
import { Menu } from './index';

describe.only('Menu', () => {
  it('Should  hidden navigator when in less 768px screen', () => {
    render(<Menu />);
    const navigator = screen.getByRole('navigation');
    expect(navigator).toHaveStyleRule('opacity', '0', {
      media: '(max-width: 768px)',
    });
    expect(navigator).toMatchSnapshot();
  });

  it('Should  show navigator when in less 768px and menu open was active', () => {
    render(<Menu />);
    const navigator = screen.getByRole('navigation');
    const openMenu = screen.getByLabelText(/Open And Close Menu/gi);
    expect(navigator).toHaveStyleRule('opacity', '0', {
      media: '(max-width: 768px)',
    });

    fireEvent.click(openMenu);

    expect(navigator).toHaveStyleRule('opacity', '1', {
      media: '(max-width: 768px)',
    });

    expect(navigator).toMatchSnapshot();
  });
});
