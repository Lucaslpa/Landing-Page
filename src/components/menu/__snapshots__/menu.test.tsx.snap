// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Should  hidden navigator when in less 768px screen 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 100%;
  max-width: 70rem;
  background-color: #FFFFFF;
  opacity: 1;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.c0 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 42.7rem;
  width: 100%;
}

.c0 ul li {
  list-style: none;
  cursor: pointer;
}

.c0 ul li a {
  font-size: 1.7rem;
  font-family: Inter;
  font-weight: 400;
  color: #7B7B7B;
}

.c0 ul li a::after {
  content: 'sad' width:100px;
  height: 0.4rem;
  background-color: black;
}

.c0 ul li:first-child a {
  color: #252160 !important;
  font-weight: 700;
}

.c1 {
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  box-sizing: border-box;
  height: 100vh;
  width: 100%;
  cursor: pointer;
  font-family: Inter;
  font-weight: bold;
  line-height: 21px;
  font-style: normal;
  color: #FFFFFF;
  background-color: #252160;
  fill: #FFFFFF;
  max-width: 13.3rem;
  max-height: 4.3rem;
  font-size: 1.5rem;
  text-align: center;
  border-radius: 0.6rem;
}

.c1 path {
  fill: #FFFFFF;
}

.c1:hover {
  background-color: #27216e;
}

@media (min-width:768px) {

}

@media (max-width:1170px) {

}

@media (min-width:1170px) {

}

@media (max-width:768px) {
  .c0 {
    position: absolute;
    opacity: 0;
    pointer-events: none;
    top: 99%;
    left: 0%;
    right: 0;
    height: 85vh;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c0 ul {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    max-width: 100%;
    margin-bottom: 4rem;
    height: auto;
  }

  .c0 ul li {
    margin: 1.4rem 0;
  }

  .c0 ul li a {
    font-size: 3rem;
  }
}

<nav
  class="c0"
>
  <ul>
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/"
      >
        Quem Somos
      </a>
    </li>
    <li>
      <a
        href="/"
      >
        Planos
      </a>
    </li>
    <li>
      <a
        href="/"
      >
        Fale conosco
      </a>
    </li>
  </ul>
  <a
    aria-label="Entrar"
    class="c1"
    href="/"
  >
    <span>
      Entrar
    </span>
  </a>
</nav>
`;

exports[`Menu Should  show navigator when in less 768px and menu open was active 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 100%;
  max-width: 70rem;
  background-color: #FFFFFF;
  opacity: 1;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.c0 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-width: 42.7rem;
  width: 100%;
}

.c0 ul li {
  list-style: none;
  cursor: pointer;
}

.c0 ul li a {
  font-size: 1.7rem;
  font-family: Inter;
  font-weight: 400;
  color: #7B7B7B;
}

.c0 ul li a::after {
  content: 'sad' width:100px;
  height: 0.4rem;
  background-color: black;
}

.c0 ul li:first-child a {
  color: #252160 !important;
  font-weight: 700;
}

.c1 {
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  box-sizing: border-box;
  height: 100vh;
  width: 100%;
  cursor: pointer;
  font-family: Inter;
  font-weight: bold;
  line-height: 21px;
  font-style: normal;
  color: #FFFFFF;
  background-color: #252160;
  fill: #FFFFFF;
  max-width: 13.3rem;
  max-height: 4.3rem;
  font-size: 1.5rem;
  text-align: center;
  border-radius: 0.6rem;
}

.c1 path {
  fill: #FFFFFF;
}

.c1:hover {
  background-color: #27216e;
}

@media (min-width:768px) {

}

@media (max-width:1170px) {

}

@media (min-width:1170px) {

}

@media (max-width:768px) {

}

@media (max-width:768px) {
  .c0 {
    position: absolute;
    opacity: 1;
    pointer-events: all;
    top: 99%;
    left: 0%;
    right: 0;
    height: 85vh;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }

  .c0 ul {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    max-width: 100%;
    margin-bottom: 4rem;
    height: auto;
  }

  .c0 ul li {
    margin: 1.4rem 0;
  }

  .c0 ul li a {
    font-size: 3rem;
  }
}

<nav
  class="c0"
  hidden=""
>
  <ul>
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/"
      >
        Quem Somos
      </a>
    </li>
    <li>
      <a
        href="/"
      >
        Planos
      </a>
    </li>
    <li>
      <a
        href="/"
      >
        Fale conosco
      </a>
    </li>
  </ul>
  <a
    aria-label="Entrar"
    class="c1"
    href="/"
  >
    <span>
      Entrar
    </span>
  </a>
</nav>
`;
